tempesta_fw/str.c

```diff
@@ -410,8 +410,48 @@ tfw_str_eq_cstr_pos(const TfwStr *str, const char *pos, const char *cstr,
 }
 EXPORT_SYMBOL(tfw_str_eq_cstr_pos);
 
+bool
+tfw_str_eq_cstr_off(const TfwStr *str, ssize_t offset, const char *cstr,
+		    int cstr_len, tfw_str_eq_flags_t flags)
+{
+	bool ret = false;
+	TfwStr t, tmp = *str;
+	TfwStr *c, *end;
+
+	BUG_ON(TFW_STR_DUP(str));
+	BUG_ON(!cstr || !cstr_len);
+
+	if (offset < 0)
```

Comment:
In fact, this is not an unlikely event considering how this function is used in comparisons. The length of the pattern is subtracted from the length of the URI path, and it may well be that the URI is shorter in an HTTP message.